client/modules/IDE/components/ConsoleInput.jsx

```diff
@@ -24,75 +24,116 @@ function ConsoleInput({ theme, dispatchConsoleEvent, fontSize }) {
       inputStyle: 'contenteditable'
     });
 
-    cmInstance.current.on('keydown', (cm, e) => {
+    cmInstance.current.getWrapperElement().style['font-size'] = `${fontSize}px`;
+
+    return () => {
+      if (cmInstance.current) {
+        cmInstance.current = null;
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    if (cmInstance.current) {
+      cmInstance.current.setOption('theme', `p5-${theme}`);
+      cmInstance.current.getWrapperElement().style[
+        'font-size'
+      ] = `${fontSize}px`;
+      cmInstance.current.refresh();
+    }
+  }, [theme, fontSize]);
+
+  useEffect(() => {
+    const handleEnterKey = (cm, e) => {
       if (e.key === 'Enter' && !e.shiftKey) {
         e.preventDefault();
         e.stopPropagation();
-        const value = cm.getValue();
-        if (value.trim(' ') === '') {
-          return false;
-        }
+
+        const value = cm.getValue().trim();
+        if (value === '') return;
+
         const messages = [
           { log: Encode({ method: 'command', data: [value] }) }
         ];
         const consoleEvent = [{ method: 'command', data: [value] }];
+
         dispatchMessage({
           type: MessageTypes.EXECUTE,
-          payload: {
-            source: 'console',
-            messages
-          }
+          payload: { source: 'console', messages }
         });
+
         dispatchConsoleEvent(consoleEvent);
         cm.setValue('');
         setCommandHistory([value, ...commandHistory]);
         setCommandCursor(-1);
-      } else if (e.key === 'ArrowUp') {
-        const lineNumber = cmInstance.current.getDoc().getCursor().line;
-        if (lineNumber !== 0) {
-          return false;
-        }
+      }
+    };
+
+    if (cmInstance.current) {
+      cmInstance.current.on('keydown', handleEnterKey);
+    }
+
+    return () => {
+      if (cmInstance.current) {
+        cmInstance.current.off('keydown', handleEnterKey);
+      }
+    };
+  }, [commandHistory, dispatchConsoleEvent]);
```

Comment:
The `off` and `on` is good.  That will prevent us having multiple event handlers.

Having `dispatchConsoleEvent` as a dependency means that this effect will re-run on every render of `Console` because the function is declared inline and is not memoized.  You could memoize it (with `useCallback`) but I would instead handle the dispatching here instead of passing a function prop.